includes/html/api_functions.inc.php

```diff
@@ -2249,6 +2249,50 @@ function add_service_for_host(\Illuminate\Http\Request $request)
     return api_error(500, 'Failed to add the service');
 }
 
+function add_parents_to_host(\Illuminate\Http\Request $request)
+{
+    $data = json_decode($request->getContent(), true);
+    $device_id = $request->route('id');
+    $parent_ids = explode(',', $data['parent_ids']);
+    if (validateDeviceIds($parent_ids) && validateDeviceIds(array($device_id)) && (!in_array($device_id, $parent_ids))) {
+        Device::find($device_id)->parents()->sync($parent_ids);
+        return api_success_noresult(201, 'Device dependencies have been saved');
+    }
+    return api_error(400, "Check your parent and device IDs");
+}
+
+function del_parents_from_host(\Illuminate\Http\Request $request)
+{
+    $device_id = $request->route('id');
+    $data = json_decode($request->getContent(), true);
+    $device = Device::find($device_id);
+    if (!validateDeviceIds(array($device_id))) {
+        return api_error(400, "Check your device ID!");
+    }
+    if (!empty($data['parent_ids'])) {
+        $parent_ids = explode(',', $data['parent_ids']);
+        if (!validateDeviceIds($parent_ids) || !Device::find($device_id)->parents()->detach($parent_ids)) {
+            return api_error(400, "Device dependency cannot be deleted check your parent IDs!");
+        }
+        return api_success_noresult(201, 'Device dependencies have been removed');
+    }
```

Comment:
do you first remove associated parent_id's and then all parent_id's? ;-)

Reply:
No, in that code if $data['parent_ids'] is empty i deleted all the parents, else i deleted only the parents sent in the request. Anyway i refactored this code andd tested it. If it passes codeview's test we can discuss it further :).

Reply:
Ok, the code seems to be ready. If we send one or more parent_ids to the function del_parents_from_host it wil remove only those parents. If you don't send any oarent_ids it will remove all parents. I tested it with my server and it works. Thanks again for your input.